From: Frank Wang <frank.wang@rock-chips.com>
Date: Tue, 16 Aug 2016 14:13:42 +0800
Subject: phy: rockchip-inno-usb2: add COMMON_CLK dependency
Git-commit: 3fa295667a70e8f2e7e0251dbaa615b01ca82b23
Patch-mainline: v4.9-rc1
References: fate#322003

On kernel builds without COMMON_CLK, the newly added rockchip-inno-usb2
driver fails to build:

drivers/phy/phy-rockchip-inno-usb2.c:124:16: error: field 'clk480m_hw'
has incomplete type
   struct clk_hw clk480m_hw;

In file included from include/linux/clk.h:16:0
                 from drivers/phy/phy-rockchip-inno-usb2.c:17:
include/linux/kernel.h:831:48: error: initialization from incompatible
pointer type [-Werror=incompatible-pointer-types]
   const typeof( ((type *)0)->member ) *__mptr = (ptr); \

  ... ...

Signed-off-by: Frank Wang <frank.wang@rock-chips.com>
Signed-off-by: Kishon Vijay Abraham I <kishon@ti.com>
Reviewed-by: Guenter Roeck <linux@roeck-us.net>
Reviewed-by: Heiko Stuebner <heiko@sntech.de>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/phy/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/phy/Kconfig b/drivers/phy/Kconfig
index 46e5536c453e..8cf6c3d0f71a 100644
--- a/drivers/phy/Kconfig
+++ b/drivers/phy/Kconfig
@@ -370,6 +370,7 @@ config PHY_ROCKCHIP_USB
 config PHY_ROCKCHIP_INNO_USB2
 	tristate "Rockchip INNO USB2PHY Driver"
 	depends on (ARCH_ROCKCHIP || COMPILE_TEST) && OF
+	depends on COMMON_CLK
 	select GENERIC_PHY
 	help
 	  Support for Rockchip USB2.0 PHY with Innosilicon IP block.
-- 
2.11.0

